### PR TITLE
feat(vue): add controlled behavior for list components

### PR DESCRIPTION
**Pull request by jyash97**
- [ ]  Add Docs

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@jyash97 I have some questions:
1. I can't see `defaultValue` implemented.
2. What is the purpose of not removing the  `defaultSelected` prop?

Make these changes [here](https://github.com/appbaseio/reactivesearch/pull/1194/files) too.

**Comment by jyash97** [comment]
@bietkul I havent changed the functioning of `defaultSelected`. In this PR and the other PR I have just added support for controlled component. For `defaultValue`  we just need to do a rename and test if everything works properly which can be a separate PR I guess.

**Comment by bietkul** [comment]
@jyash97 Although we can remove the `defaultSelected` prop later but The `defaultValue` prop is the part of controlled behavior, you should include that in this PR.
So do these changes:
1. Implement `defaultValue`.
2. Mark `defaultSelected` as deprecated & throw a warning if used.

File: packages/vue/src/components/list/MultiDropdownList.jsx [hunk 1]
```diff
@@ -87,7 +89,11 @@ const MultiDropdownList = {
 
 		if (this.selectedValue) {
 			this.setValue(this.selectedValue, true);
+		} else if (this.$props.defaultValue) {
+			this.setValue(this.$props.defaultValue, true);
 		} else if (this.$props.defaultSelected) {
+			/* TODO: Remove this before next release */
+			console.warn("defaultSelected prop will be deprecated in the next release. Please replace it with defaultValue before upgrading to the next major version.");
```
Comment (bietkul) [inline packages/vue/src/components/list/MultiDropdownList.jsx]: This warning message gets repeated in many places, @jyash97 can you put it out at in some util method.

File: packages/vue/src/components/list/MultiDropdownList.jsx [hunk 2]
```diff
@@ -257,7 +280,7 @@ const MultiDropdownList = {
 				finalValues = value;
 				currentValue = {};
 
-				if (value) {
+				if (value && value.length) {
```
Comment (bietkul) [inline packages/vue/src/components/list/MultiDropdownList.jsx]: Why there is a length check?
Are you trying to make sure that the value is an `Array`?
If yes then use `Array.isArray()` instead.

File: packages/vue/src/components/list/MultiDropdownList.jsx [hunk 3]
```diff
@@ -151,6 +157,12 @@ const MultiDropdownList = {
 		defaultSelected(newVal) {
 			this.setValue(newVal, true);
 		},
+		defaultValue(newVal) {
+			this.setValue(newVal, true);
+		},
+		value(newVal) {
+			this.setValue(newVal, true);
```
Comment (bietkul) [inline packages/vue/src/components/list/MultiDropdownList.jsx]: Use `isEqual` to compare the values.

File: packages/vue/src/components/list/SingleDropdownList.jsx [hunk 4]
```diff
@@ -135,6 +143,12 @@ const SingleDropdownList = {
 		defaultSelected(newVal) {
 			this.setValue(newVal);
 		},
+		defaultValue(newVal) {
+			this.setValue(newVal);
+		},
+		value(newVal) {
```
Comment (bietkul) [inline packages/vue/src/components/list/SingleDropdownList.jsx]: Check the above comments.

File: packages/vue/src/components/list/MultiList.jsx [hunk 5]
```diff
@@ -297,7 +315,7 @@ const MultiList = {
 				finalValues = value;
 				currentValue = {};
 
-				if (value) {
+				if (value && value.length) {
```
Comment (bietkul) [inline packages/vue/src/components/list/MultiList.jsx:319]: Check the above comments.

File: packages/vue/src/components/list/SingleList.jsx [hunk 6]
```diff
@@ -114,6 +122,12 @@ const SingleList = {
 		defaultSelected(newVal) {
 			this.setValue(newVal);
 		},
+		defaultValue(newVal) {
+			this.setValue(newVal);
+		},
+		value(newVal) {
```
Comment (bietkul) [inline packages/vue/src/components/list/SingleList.jsx]: Check the above comments.

**Review by bietkul — CHANGES_REQUESTED** [review]
@jyash97 Please check the comments, the same issues are present in multiple files.
Also, update [this](https://github.com/appbaseio/reactivesearch/pull/1194/files) PR.

**Review by bietkul — APPROVED** [review]

**Pull request merged**